graph_weather/data/anemoi_dataloder.py

```diff
@@ -0,0 +1,185 @@
+            else:
+                self.data = self.dataset
+            logging.info(f"Successfully loaded Anemoi dataset: {dataset_name}")
+        except Exception as e:
+            raise RuntimeError(
+                f"Failed to load Anemoi dataset '{dataset_name}': {e}. "
+                "Please ensure the dataset is available and properly configured."
+            )
+
+        # Validate that we have the required features
+        missing_features = [f for f in self.features if f not in self.data.data_vars]
+        if missing_features:
+            available_features = list(self.data.data_vars.keys())
+            raise ValueError(
+                f"Features {missing_features} not found in dataset. "
+                f"Available features: {available_features}"
+            )
+
+        # Get grid information - try multiple coordinate name variations
+        coord_names = ["latitude", "lat", "y"]
+        self.grid_lat = None
+        for name in coord_names:
+            if name in self.data.coords:
+                self.grid_lat = self.data.coords[name]
+                break
+
+        coord_names = ["longitude", "lon", "x"]
+        self.grid_lon = None
+        for name in coord_names:
+            if name in self.data.coords:
+                self.grid_lon = self.data.coords[name]
+                break
+
+        if self.grid_lat is None or self.grid_lon is None:
+            available_coords = list(self.data.coords.keys())
+            raise ValueError(
+                f"Could not find latitude/longitude coordinates in dataset. "
+                f"Available coordinates: {available_coords}"
+            )
+
+        self.num_lat = len(self.grid_lat)
+        self.num_lon = len(self.grid_lon)
+
+        # Initialize normalization parameters
+        self.means, self.stds = self._init_normalization()
+
+    def _init_normalization(self):
+        """Initialize normalization parameters"""
+        means = {}
+        stds = {}
+
+        for feature in self.features:
+            if feature in self.data.data_vars:
+                data_values = self.data[feature].values
+                means[feature] = np.nanmean(data_values)
+                stds[feature] = np.nanstd(data_values)
+            else:
+                logging.warning(f"Feature {feature} not found, using default normalization")
+                means[feature] = 0.0
+                stds[feature] = 1.0
+
+        return means, stds
+
+    def _normalize(self, data, feature):
+        """Normalize data using feature-specific statistics"""
+        return (data - self.means[feature]) / (self.stds[feature] + 1e-6)
+
+    def _generate_clock_features(self, data_time):
+        """Generate time features following GenCast pattern"""
+        if hasattr(data_time, "values"):
+            timestamp = pd.Timestamp(data_time.values)
+        else:
+            timestamp = data_time
+
+        day_of_year = timestamp.dayofyear / 365.0
```

Comment:
Please use 366 for leap years to stay within 0-1
```suggestion
        day_of_year = timestamp.dayofyear / 366.0
```